app/services/open_library/response_adapter.rb

```diff
@@ -0,0 +1,33 @@
+module OpenLibrary
+  class ResponseAdapter
+    def initialize(body, isbn, status)
+      @body = body
+      @isbn = isbn
+      @status = status
+    end
+
+    def book_response
+      {
+        "isbn": @isbn,
+        "title": book_body['title'],
+        "subtitle": book_body['subtitle'],
+        "number_of_pages": book_body['number_of_pages'],
+        "authors": book_body['authors']
+      }
+    end
+
+    def success?
+      @status >= 200 && @status < 400 && @body.present?
```

Comment:
it's better to check for 2.. response codes. 3xx could be a redirect
`@status.to_s.match?(/^2..$/)`